org.eclipse.scout.rt.jackson/src/main/java/org/eclipse/scout/rt/jackson/dataobject/ScoutDataObjectSerializerProvider.java

```diff
@@ -64,6 +64,9 @@ public JsonSerializer<?> findSerializer(ScoutDataObjectModuleContext moduleConte
     else if (ObjectUtility.isOneOf(rawClass, DoList.class, DoSet.class, DoCollection.class)) {
       return new DoCollectionSerializer<>(moduleContext, type);
     }
+    else if (IDataObject.class.isAssignableFrom(rawClass)) {
```

Comment:
Why is the own serializer required?
When gen.writeObjectField(attributeName, obj); is called in DoEntitySerializer (due to the modified if clause), isn't the serializer determined based on the value type anyway, which is either IDoEntity or one of the collections above?

Reply:
The DataObjectSerializer is required if a mapper for IDataObject.class is requested explicitly.
See test: 
org.eclipse.scout.rt.jackson.dataobject.JsonDataObjectsSerializationTest#testSerializeDeserializeDataObject_DoEntity
org.eclipse.scout.rt.jackson.dataobject.JsonDataObjectsSerializationTest#testSerializeDeserializeDataObject_DoList

Unfortunately I was not able to write a test to force an error if this serializer is missing using a DoEntity and a DoValue<IDataObject> attribute, since in this case your input is correct and DoEntitySerializer handles the case.